src/services/registration.js

```diff
@@ -0,0 +1,32 @@
+// const regModel = require('../models/registrationExample');
+const checkName = require('../services/checkUserName');
+const checkPsw = require('../services/checkPassword');
+const dbCheck = require('../services/regCheckDB');
+const users = require('../models/userDB');
+
+const createUser = item => new Promise((resolve) => {
+  const temp = {
+    username: '',
+    password: '',
+  };
+  if (checkName.CheckUserName(item.username) && checkPsw.CheckPassword(item.password)) {
+    if (item.password === item.confirmPsw) {
+      if (!dbCheck.containsUser(item, users.userDB)) {
+        temp.username = item.username;
+        temp.password = item.password;
+        users.userDB.push(temp);
```

Comment:
This should be done like push({username: item.username, password: item.password}).
Or you could dismiss the fact that you pass extra information and just call push(item).

Using the temp variable is unfortunate in some aspects, like the naming and it's created very far from the point of usage. Also you create it even if you won't use it for example if the username is empty.

Reply:
I made it better but I still have the variable. I would like to send the exact data, not everything.